UI keeps stopping on Android 11

Long-press - Shuffle All seems to work.

Logcat, in case someone who knows Java feels like fixing it, since the dev seems to have disappeared.

```
2020-08-25 20:09:37.903 12450-12450/? I/ramophone.debu: Late-enabling -Xcheck:jni
2020-08-25 20:09:37.932 12450-12450/? I/ramophone.debu: Unquickening 12 vdex files!
2020-08-25 20:09:38.250 12450-12450/com.kabouzeid.gramophone.debug D/ApplicationLoaders: Returning zygote-cached class loader: /system/framework/android.test.base.jar
2020-08-25 20:09:38.425 12450-12450/com.kabouzeid.gramophone.debug D/NetworkSecurityConfig: No Network Security Config specified, using platform default
2020-08-25 20:09:38.426 12450-12450/com.kabouzeid.gramophone.debug D/NetworkSecurityConfig: No Network Security Config specified, using platform default
2020-08-25 20:09:38.735 12450-12450/com.kabouzeid.gramophone.debug W/ramophone.debu: Accessing hidden method Landroid/view/View;->computeFitSystemWindows(Landroid/graphics/Rect;Landroid/graphics/Rect;)Z (greylist, reflection, allowed)
2020-08-25 20:09:38.736 12450-12450/com.kabouzeid.gramophone.debug W/ramophone.debu: Accessing hidden method Landroid/view/ViewGroup;->makeOptionalFitsSystemWindows()V (greylist, reflection, allowed)
2020-08-25 20:09:39.299 12450-12450/com.kabouzeid.gramophone.debug W/ramophone.debu: Accessing hidden method Landroid/graphics/Canvas;->save(I)I (greylist, linking, allowed)
2020-08-25 20:09:39.385 12450-12514/com.kabouzeid.gramophone.debug I/AdrenoGLES-0: QUALCOMM build                   : 02fe52e, Ie73904e3bd
    Build Date                       : 06/27/20
    OpenGL ES Shader Compiler Version: EV031.31.04.00
    Local Branch                     : gfx-adreno.lnx.2.0
    Remote Branch                    : quic/gfx-adreno.lnx.2.0
    Remote Branch                    : NONE
    Reconstruct Branch               : NOTHING
2020-08-25 20:09:39.385 12450-12514/com.kabouzeid.gramophone.debug I/AdrenoGLES-0: Build Config                     : S L 10.0.5 AArch64
2020-08-25 20:09:39.385 12450-12514/com.kabouzeid.gramophone.debug I/AdrenoGLES-0: Driver Path                      : /vendor/lib64/egl/libGLESv2_adreno.so
2020-08-25 20:09:39.403 12450-12514/com.kabouzeid.gramophone.debug I/AdrenoGLES-0: PFP: 0x005ff112, ME: 0x005ff066
2020-08-25 20:09:39.404 12450-12514/com.kabouzeid.gramophone.debug W/AdrenoUtils: <ReadGpuID_from_sysfs:197>: Failed to open /sys/class/kgsl/kgsl-3d0/gpu_model
2020-08-25 20:09:39.404 12450-12514/com.kabouzeid.gramophone.debug W/AdrenoUtils: <ReadGpuID:221>: Failed to read chip ID from gpu_model. Fallback to use the GSL path
2020-08-25 20:09:39.489 12450-12514/com.kabouzeid.gramophone.debug I/Gralloc4: mapper 4.x is not supported
2020-08-25 20:09:39.490 12450-12514/com.kabouzeid.gramophone.debug W/Gralloc3: mapper 3.x is not supported
2020-08-25 20:09:39.593 12450-12529/com.kabouzeid.gramophone.debug E/AndroidRuntime: FATAL EXCEPTION: ModernAsyncTask #4
    Process: com.kabouzeid.gramophone.debug, PID: 12450
    java.lang.RuntimeException: An error occurred while executing doInBackground()
        at androidx.loader.content.ModernAsyncTask$3.done(ModernAsyncTask.java:164)
        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
        at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
        at java.util.concurrent.FutureTask.run(FutureTask.java:271)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:923)
     Caused by: java.lang.UnsupportedOperationException: Unknown or unsupported URL: content://media/external/audio/genres/-941897654/members
        at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:174)
        at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:142)
        at android.content.ContentProviderProxy.query(ContentProviderNative.java:472)
        at android.content.ContentResolver.query(ContentResolver.java:1183)
        at android.content.ContentResolver.query(ContentResolver.java:1115)
        at android.content.ContentResolver.query(ContentResolver.java:1071)
        at com.kabouzeid.gramophone.loader.GenreLoader.makeGenreSongCursor(GenreLoader.java:64)
        at com.kabouzeid.gramophone.loader.GenreLoader.getSongs(GenreLoader.java:25)
        at com.kabouzeid.gramophone.loader.GenreLoader.getGenreFromCursor(GenreLoader.java:57)
        at com.kabouzeid.gramophone.loader.GenreLoader.getGenresFromCursor(GenreLoader.java:34)
        at com.kabouzeid.gramophone.loader.GenreLoader.getAllGenres(GenreLoader.java:20)
        at com.kabouzeid.gramophone.ui.fragments.mainactivity.library.pager.GenresFragment$AsyncGenreLoader.loadInBackground(GenresFragment.java:76)
        at com.kabouzeid.gramophone.ui.fragments.mainactivity.library.pager.GenresFragment$AsyncGenreLoader.loadInBackground(GenresFragment.java:69)
        at androidx.loader.content.AsyncTaskLoader.onLoadInBackground(AsyncTaskLoader.java:307)
        at androidx.loader.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:60)
        at androidx.loader.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:48)
        at androidx.loader.content.ModernAsyncTask$2.call(ModernAsyncTask.java:141)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:923) 
2020-08-25 20:09:39.632 12450-12529/com.kabouzeid.gramophone.debug I/Process: Sending signal. PID: 12450 SIG: 9

```

--- a/app/src/main/java/com/kabouzeid/gramophone/dialogs/CreatePlaylistDialog.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/dialogs/CreatePlaylistDialog.java
@@ -61,7 +61,7 @@ public class CreatePlaylistDialog extends DialogFragment {
                     final String name = charSequence.toString().trim();
                     if (!name.isEmpty()) {
                         if (!PlaylistsUtil.doesPlaylistExist(getActivity(), name)) {
-                            final int playlistId = PlaylistsUtil.createPlaylist(getActivity(), name);
+                            final long playlistId = PlaylistsUtil.createPlaylist(getActivity(), name);
                             if (getActivity() != null) {
                                 //noinspection unchecked
                                 List<Song> songs = getArguments().getParcelableArrayList(SONGS);

--- a/app/src/main/java/com/kabouzeid/gramophone/loader/AlbumLoader.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/loader/AlbumLoader.java
@@ -45,7 +45,7 @@ public class AlbumLoader {
     }
 
     @NonNull
-    public static Album getAlbum(@NonNull final Context context, int albumId) {
+    public static Album getAlbum(@NonNull final Context context, long albumId) {
         List<Song> songs = SongLoader.getSongs(SongLoader.makeSongCursor(context, AudioColumns.ALBUM_ID + "=?", new String[]{String.valueOf(albumId)}, getSongLoaderSortOrder(context)));
         Album album = new Album(songs);
         sortSongsByTrackNumber(album);
@@ -66,7 +66,7 @@ public class AlbumLoader {
         return albums;
     }
 
-    private static Album getOrCreateAlbum(List<Album> albums, int albumId) {
+    private static Album getOrCreateAlbum(List<Album> albums, long albumId) {
         for (Album album : albums) {
             if (!album.songs.isEmpty() && album.songs.get(0).albumId == albumId) {
                 return album;

--- a/app/src/main/java/com/kabouzeid/gramophone/loader/ArtistLoader.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/loader/ArtistLoader.java
@@ -44,7 +44,7 @@ public class ArtistLoader {
     }
 
     @NonNull
-    public static Artist getArtist(@NonNull final Context context, int artistId) {
+    public static Artist getArtist(@NonNull final Context context, long artistId) {
         List<Song> songs = SongLoader.getSongs(SongLoader.makeSongCursor(
                 context,
                 AudioColumns.ARTIST_ID + "=?",
@@ -65,7 +65,7 @@ public class ArtistLoader {
         return artists;
     }
 
-    private static Artist getOrCreateArtist(List<Artist> artists, int artistId) {
+    private static Artist getOrCreateArtist(List<Artist> artists, long artistId) {
         for (Artist artist : artists) {
             if (!artist.albums.isEmpty() && !artist.albums.get(0).songs.isEmpty() && artist.albums.get(0).songs.get(0).artistId == artistId) {
                 return artist;

--- a/app/src/main/java/com/kabouzeid/gramophone/loader/GenreLoader.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/loader/GenreLoader.java
@@ -21,7 +21,7 @@ public class GenreLoader {
     }
 
     @NonNull
-    public static List<Song> getSongs(@NonNull final Context context, final int genreId) {
+    public static List<Song> getSongs(@NonNull final Context context, final long genreId) {
         return SongLoader.getSongs(makeGenreSongCursor(context, genreId));
     }
 
@@ -52,14 +52,14 @@ public class GenreLoader {
 
     @NonNull
     private static Genre getGenreFromCursor(@NonNull final Context context, @NonNull final Cursor cursor) {
-        final int id = cursor.getInt(0);
+        final long id = cursor.getLong(0);
         final String name = cursor.getString(1);
         final int songs = getSongs(context, id).size();
         return new Genre(id, name, songs);
     }
 
     @Nullable
-    private static Cursor makeGenreSongCursor(@NonNull final Context context, int genreId) {
+    private static Cursor makeGenreSongCursor(@NonNull final Context context, long genreId) {
         try {
             return context.getContentResolver().query(
                     Genres.Members.getContentUri("external", genreId),

--- a/app/src/main/java/com/kabouzeid/gramophone/loader/PlaylistLoader.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/loader/PlaylistLoader.java
@@ -21,7 +21,7 @@ public class PlaylistLoader {
     }
 
     @NonNull
-    public static Playlist getPlaylist(@NonNull final Context context, final int playlistId) {
+    public static Playlist getPlaylist(@NonNull final Context context, final long playlistId) {
         return getPlaylist(makePlaylistCursor(
                 context,
                 BaseColumns._ID + "=?",

--- a/app/src/main/java/com/kabouzeid/gramophone/loader/PlaylistSongLoader.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/loader/PlaylistSongLoader.java
@@ -14,7 +14,7 @@ import java.util.List;
 public class PlaylistSongLoader {
 
     @NonNull
-    public static List<PlaylistSong> getPlaylistSongList(@NonNull final Context context, final int playlistId) {
+    public static List<PlaylistSong> getPlaylistSongList(@NonNull final Context context, final long playlistId) {
         List<PlaylistSong> songs = new ArrayList<>();
         Cursor cursor = makePlaylistSongCursor(context, playlistId);
 
@@ -30,7 +30,7 @@ public class PlaylistSongLoader {
     }
 
     @NonNull
-    private static PlaylistSong getPlaylistSongFromCursorImpl(@NonNull Cursor cursor, int playlistId) {
+    private static PlaylistSong getPlaylistSongFromCursorImpl(@NonNull Cursor cursor, long playlistId) {
         final long id = cursor.getLong(0);
         final String title = cursor.getString(1);
         final int trackNumber = cursor.getInt(2);
@@ -47,7 +47,7 @@ public class PlaylistSongLoader {
         return new PlaylistSong(id, title, trackNumber, year, duration, data, dateModified, albumId, albumName, artistId, artistName, playlistId, idInPlaylist);
     }
 
-    public static Cursor makePlaylistSongCursor(@NonNull final Context context, final int playlistId) {
+    public static Cursor makePlaylistSongCursor(@NonNull final Context context, final long playlistId) {
         try {
             return context.getContentResolver().query(
                     MediaStore.Audio.Playlists.Members.getContentUri("external", playlistId),

--- a/app/src/main/java/com/kabouzeid/gramophone/loader/PlaylistSongLoader.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/loader/PlaylistSongLoader.java
@@ -31,18 +31,18 @@ public class PlaylistSongLoader {
 
     @NonNull
     private static PlaylistSong getPlaylistSongFromCursorImpl(@NonNull Cursor cursor, int playlistId) {
-        final int id = cursor.getInt(0);
+        final long id = cursor.getLong(0);
         final String title = cursor.getString(1);
         final int trackNumber = cursor.getInt(2);
         final int year = cursor.getInt(3);
         final long duration = cursor.getLong(4);
         final String data = cursor.getString(5);
         final int dateModified = cursor.getInt(6);
-        final int albumId = cursor.getInt(7);
+        final long albumId = cursor.getLong(7);
         final String albumName = cursor.getString(8);
-        final int artistId = cursor.getInt(9);
+        final long artistId = cursor.getLong(9);
         final String artistName = cursor.getString(10);
-        final int idInPlaylist = cursor.getInt(11);
+        final long idInPlaylist = cursor.getLong(11);
 
         return new PlaylistSong(id, title, trackNumber, year, duration, data, dateModified, albumId, albumName, artistId, artistName, playlistId, idInPlaylist);
     }

--- a/app/src/main/java/com/kabouzeid/gramophone/loader/SongLoader.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/loader/SongLoader.java
@@ -82,16 +82,16 @@ public class SongLoader {
 
     @NonNull
     private static Song getSongFromCursorImpl(@NonNull Cursor cursor) {
-        final int id = cursor.getInt(0);
+        final long id = cursor.getLong(0);
         final String title = cursor.getString(1);
         final int trackNumber = cursor.getInt(2);
         final int year = cursor.getInt(3);
         final long duration = cursor.getLong(4);
         final String data = cursor.getString(5);
         final long dateModified = cursor.getLong(6);
-        final int albumId = cursor.getInt(7);
+        final long albumId = cursor.getLong(7);
         final String albumName = cursor.getString(8);
-        final int artistId = cursor.getInt(9);
+        final long artistId = cursor.getLong(9);
         final String artistName = cursor.getString(10);
 
         return new Song(id, title, trackNumber, year, duration, data, dateModified, albumId, albumName, artistId, artistName);

--- a/app/src/main/java/com/kabouzeid/gramophone/model/AbsCustomPlaylist.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/model/AbsCustomPlaylist.java
@@ -12,7 +12,7 @@ import java.util.List;
  */
 
 public abstract class AbsCustomPlaylist extends Playlist {
-    public AbsCustomPlaylist(int id, String name) {
+    public AbsCustomPlaylist(long id, String name) {
         super(id, name);
     }
 

--- a/app/src/main/java/com/kabouzeid/gramophone/model/Album.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/model/Album.java
@@ -21,7 +21,7 @@ public class Album implements Parcelable {
         this.songs = new ArrayList<>();
     }
 
-    public int getId() {
+    public long getId() {
         return safeGetFirstSong().albumId;
     }
 
@@ -29,7 +29,7 @@ public class Album implements Parcelable {
         return safeGetFirstSong().albumName;
     }
 
-    public int getArtistId() {
+    public long getArtistId() {
         return safeGetFirstSong().artistId;
     }
 

--- a/app/src/main/java/com/kabouzeid/gramophone/model/Artist.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/model/Artist.java
@@ -25,7 +25,7 @@ public class Artist implements Parcelable {
         this.albums = new ArrayList<>();
     }
 
-    public int getId() {
+    public long getId() {
         return safeGetFirstAlbum().getArtistId();
     }
 

--- a/app/src/main/java/com/kabouzeid/gramophone/model/Genre.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/model/Genre.java
@@ -4,11 +4,11 @@ import android.os.Parcel;
 import android.os.Parcelable;
 
 public class Genre implements Parcelable {
-    public final int id;
+    public final long id;
     public final String name;
     public final int songCount;
 
-    public Genre(final int id, final String name, final int songCount) {
+    public Genre(final long id, final String name, final int songCount) {
         this.id = id;
         this.name = name;
         this.songCount = songCount;
@@ -28,10 +28,10 @@ public class Genre implements Parcelable {
 
     @Override
     public int hashCode() {
-        int result = id;
+        long result = id;
         result = 31 * result + name.hashCode();
         result = 31 * result + songCount;
-        return result;
+        return (int)result;
     }
 
     @Override
@@ -50,13 +50,13 @@ public class Genre implements Parcelable {
 
     @Override
     public void writeToParcel(Parcel dest, int flags) {
-        dest.writeInt(this.id);
+        dest.writeLong(this.id);
         dest.writeString(this.name);
         dest.writeInt(this.songCount);
     }
 
     protected Genre(Parcel in) {
-        this.id = in.readInt();
+        this.id = in.readLong();
         this.name = in.readString();
         this.songCount = in.readInt();
     }

--- a/app/src/main/java/com/kabouzeid/gramophone/model/Playlist.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/model/Playlist.java
@@ -8,10 +8,10 @@ import androidx.annotation.NonNull;
  * @author Karim Abou Zeid (kabouzeid)
  */
 public class Playlist implements Parcelable {
-    public final int id;
+    public final long id;
     public final String name;
 
-    public Playlist(final int id, final String name) {
+    public Playlist(final long id, final String name) {
         this.id = id;
         this.name = name;
     }
@@ -35,7 +35,7 @@ public class Playlist implements Parcelable {
 
     @Override
     public int hashCode() {
-        int result = id;
+        int result = (int)id;
         result = 31 * result + (name != null ? name.hashCode() : 0);
         return result;
     }
@@ -56,12 +56,12 @@ public class Playlist implements Parcelable {
 
     @Override
     public void writeToParcel(Parcel dest, int flags) {
-        dest.writeInt(this.id);
+        dest.writeLong(this.id);
         dest.writeString(this.name);
     }
 
     protected Playlist(Parcel in) {
-        this.id = in.readInt();
+        this.id = in.readLong();
         this.name = in.readString();
     }
 

--- a/app/src/main/java/com/kabouzeid/gramophone/model/PlaylistSong.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/model/PlaylistSong.java
@@ -5,10 +5,10 @@ import android.os.Parcel;
 public class PlaylistSong extends Song {
     public static final PlaylistSong EMPTY_PLAYLIST_SONG = new PlaylistSong(-1, "", -1, -1, -1, "", -1, -1, "", -1, "", -1, -1);
 
-    public final int playlistId;
-    public final int idInPlayList;
+    public final long playlistId;
+    public final long idInPlayList;
 
-    public PlaylistSong(int id, String title, int trackNumber, int year, long duration, String data, int dateModified, int albumId, String albumName, int artistId, String artistName, final int playlistId, final int idInPlayList) {
+    public PlaylistSong(long id, String title, int trackNumber, int year, long duration, String data, int dateModified, long albumId, String albumName, long artistId, String artistName, final long playlistId, final long idInPlayList) {
         super(id, title, trackNumber, year, duration, data, dateModified, albumId, albumName, artistId, artistName);
         this.playlistId = playlistId;
         this.idInPlayList = idInPlayList;
@@ -30,8 +30,8 @@ public class PlaylistSong extends Song {
     @Override
     public int hashCode() {
         int result = super.hashCode();
-        result = 31 * result + playlistId;
-        result = 31 * result + idInPlayList;
+        result = 31 * result + (int)playlistId;
+        result = 31 * result + (int)idInPlayList;
         return result;
     }
 
@@ -53,14 +53,14 @@ public class PlaylistSong extends Song {
     @Override
     public void writeToParcel(Parcel dest, int flags) {
         super.writeToParcel(dest, flags);
-        dest.writeInt(this.playlistId);
-        dest.writeInt(this.idInPlayList);
+        dest.writeLong(this.playlistId);
+        dest.writeLong(this.idInPlayList);
     }
 
     protected PlaylistSong(Parcel in) {
         super(in);
-        this.playlistId = in.readInt();
-        this.idInPlayList = in.readInt();
+        this.playlistId = in.readLong();
+        this.idInPlayList = in.readLong();
     }
 
     public static final Creator<PlaylistSong> CREATOR = new Creator<PlaylistSong>() {

--- a/app/src/main/java/com/kabouzeid/gramophone/model/Song.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/model/Song.java
@@ -9,19 +9,19 @@ import android.os.Parcelable;
 public class Song implements Parcelable {
     public static final Song EMPTY_SONG = new Song(-1, "", -1, -1, -1, "", -1, -1, "", -1, "");
 
-    public final int id;
+    public final long id;
     public final String title;
     public final int trackNumber;
     public final int year;
     public final long duration;
     public final String data;
     public final long dateModified;
-    public final int albumId;
+    public final long albumId;
     public final String albumName;
-    public final int artistId;
+    public final long artistId;
     public final String artistName;
 
-    public Song(int id, String title, int trackNumber, int year, long duration, String data, long dateModified, int albumId, String albumName, int artistId, String artistName) {
+    public Song(long id, String title, int trackNumber, int year, long duration, String data, long dateModified, long albumId, String albumName, long artistId, String artistName) {
         this.id = id;
         this.title = title;
         this.trackNumber = trackNumber;
@@ -59,16 +59,16 @@ public class Song implements Parcelable {
 
     @Override
     public int hashCode() {
-        int result = id;
+        int result = (int)id;
         result = 31 * result + (title != null ? title.hashCode() : 0);
         result = 31 * result + trackNumber;
         result = 31 * result + year;
         result = 31 * result + (int) (duration ^ (duration >>> 32));
         result = 31 * result + (data != null ? data.hashCode() : 0);
         result = 31 * result + (int) (dateModified ^ (dateModified >>> 32));
-        result = 31 * result + albumId;
+        result = 31 * result + (int)albumId;
         result = 31 * result + (albumName != null ? albumName.hashCode() : 0);
-        result = 31 * result + artistId;
+        result = 31 * result + (int)artistId;
         result = 31 * result + (artistName != null ? artistName.hashCode() : 0);
         return result;
     }
@@ -98,30 +98,30 @@ public class Song implements Parcelable {
 
     @Override
     public void writeToParcel(Parcel dest, int flags) {
-        dest.writeInt(this.id);
+        dest.writeLong(this.id);
         dest.writeString(this.title);
         dest.writeInt(this.trackNumber);
         dest.writeInt(this.year);
         dest.writeLong(this.duration);
         dest.writeString(this.data);
         dest.writeLong(this.dateModified);
-        dest.writeInt(this.albumId);
+        dest.writeLong(this.albumId);
         dest.writeString(this.albumName);
-        dest.writeInt(this.artistId);
+        dest.writeLong(this.artistId);
         dest.writeString(this.artistName);
     }
 
     protected Song(Parcel in) {
-        this.id = in.readInt();
+        this.id = in.readLong();
         this.title = in.readString();
         this.trackNumber = in.readInt();
         this.year = in.readInt();
         this.duration = in.readLong();
         this.data = in.readString();
         this.dateModified = in.readLong();
-        this.albumId = in.readInt();
+        this.albumId = in.readLong();
         this.albumName = in.readString();
-        this.artistId = in.readInt();
+        this.artistId = in.readLong();
         this.artistName = in.readString();
     }
 

--- a/app/src/main/java/com/kabouzeid/gramophone/provider/MusicPlaybackQueueStore.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/provider/MusicPlaybackQueueStore.java
@@ -42,7 +42,7 @@ public class MusicPlaybackQueueStore extends SQLiteOpenHelper {
     public static final String DATABASE_NAME = "music_playback_state.db";
     public static final String PLAYING_QUEUE_TABLE_NAME = "playing_queue";
     public static final String ORIGINAL_PLAYING_QUEUE_TABLE_NAME = "original_playing_queue";
-    private static final int VERSION = 3;
+    private static final int VERSION = 4;
 
     /**
      * Constructor of <code>MusicPlaybackState</code>

--- a/app/src/main/java/com/kabouzeid/gramophone/provider/MusicPlaybackQueueStore.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/provider/MusicPlaybackQueueStore.java
@@ -67,7 +67,7 @@ public class MusicPlaybackQueueStore extends SQLiteOpenHelper {
         builder.append("(");
 
         builder.append(BaseColumns._ID);
-        builder.append(" INT NOT NULL,");
+        builder.append(" LONG NOT NULL,");
 
         builder.append(AudioColumns.TITLE);
         builder.append(" STRING NOT NULL,");
@@ -88,13 +88,13 @@ public class MusicPlaybackQueueStore extends SQLiteOpenHelper {
         builder.append(" LONG NOT NULL,");
 
         builder.append(AudioColumns.ALBUM_ID);
-        builder.append(" INT NOT NULL,");
+        builder.append(" LONG NOT NULL,");
 
         builder.append(AudioColumns.ALBUM);
         builder.append(" STRING NOT NULL,");
 
         builder.append(AudioColumns.ARTIST_ID);
-        builder.append(" INT NOT NULL,");
+        builder.append(" LONG NOT NULL,");
 
         builder.append(AudioColumns.ARTIST);
         builder.append(" STRING NOT NULL);");

--- a/app/src/main/java/com/kabouzeid/gramophone/service/MusicService.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/service/MusicService.java
@@ -1001,7 +1001,7 @@ public class MusicService extends Service implements SharedPreferences.OnSharedP
                 break;
             case SHUFFLE_MODE_NONE:
                 this.shuffleMode = shuffleMode;
-                int currentSongId = getCurrentSong().id;
+                long currentSongId = getCurrentSong().id;
                 playingQueue = new ArrayList<>(originalPlayingQueue);
                 int newPosition = 0;
                 for (Song song : getPlayingQueue()) {

--- a/app/src/main/java/com/kabouzeid/gramophone/service/MusicService.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/service/MusicService.java
@@ -16,6 +16,7 @@ import android.graphics.drawable.Drawable;
 import android.media.AudioManager;
 import android.media.audiofx.AudioEffect;
 import android.media.session.MediaSession;
+import android.net.Uri;
 import android.os.Binder;
 import android.os.Build;
 import android.os.Handler;
@@ -222,7 +223,6 @@ public class MusicService extends Service implements SharedPreferences.OnSharedP
         getContentResolver().registerContentObserver(MediaStore.Audio.Genres.INTERNAL_CONTENT_URI, true, mediaStoreObserver);
         getContentResolver().registerContentObserver(MediaStore.Audio.Playlists.INTERNAL_CONTENT_URI, true, mediaStoreObserver);
 
-
         PreferenceUtil.getInstance(this).registerOnSharedPreferenceChangedListener(this);
 
         restoreState();
@@ -230,6 +230,8 @@ public class MusicService extends Service implements SharedPreferences.OnSharedP
         mediaSession.setActive(true);
 
         sendBroadcast(new Intent("com.kabouzeid.gramophone.PHONOGRAPH_MUSIC_SERVICE_CREATED"));
+
+        mediaStoreObserver.onChange(true);
     }
 
     private AudioManager getAudioManager() {
@@ -1347,6 +1349,7 @@ public class MusicService extends Service implements SharedPreferences.OnSharedP
             // actually call refresh when the delayed callback fires
             // do not send a sticky broadcast here
             handleAndSendChangeInternal(MEDIA_STORE_CHANGED);
+            onChange(true);
         }
     }
 

--- a/app/src/main/java/com/kabouzeid/gramophone/service/MusicService.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/service/MusicService.java
@@ -230,8 +230,6 @@ public class MusicService extends Service implements SharedPreferences.OnSharedP
         mediaSession.setActive(true);
 
         sendBroadcast(new Intent("com.kabouzeid.gramophone.PHONOGRAPH_MUSIC_SERVICE_CREATED"));
-
-        mediaStoreObserver.onChange(true);
     }
 
     private AudioManager getAudioManager() {
@@ -1349,7 +1347,6 @@ public class MusicService extends Service implements SharedPreferences.OnSharedP
             // actually call refresh when the delayed callback fires
             // do not send a sticky broadcast here
             handleAndSendChangeInternal(MEDIA_STORE_CHANGED);
-            onChange(true);
         }
     }
 

--- a/app/src/main/java/com/kabouzeid/gramophone/ui/activities/AlbumDetailActivity.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/ui/activities/AlbumDetailActivity.java
@@ -430,7 +430,7 @@ public class AlbumDetailActivity extends AbsSlidingMusicPanelActivity implements
 
     @Override
     public Loader<Album> onCreateLoader(int id, Bundle args) {
-        return new AsyncAlbumLoader(this, args.getInt(EXTRA_ALBUM_ID));
+        return new AsyncAlbumLoader(this, args.getLong(EXTRA_ALBUM_ID));
     }
 
     @Override
@@ -445,9 +445,9 @@ public class AlbumDetailActivity extends AbsSlidingMusicPanelActivity implements
     }
 
     private static class AsyncAlbumLoader extends WrappedAsyncTaskLoader<Album> {
-        private final int albumId;
+        private final long albumId;
 
-        public AsyncAlbumLoader(Context context, int albumId) {
+        public AsyncAlbumLoader(Context context, long albumId) {
             super(context);
             this.albumId = albumId;
         }

--- a/app/src/main/java/com/kabouzeid/gramophone/ui/activities/ArtistDetailActivity.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/ui/activities/ArtistDetailActivity.java
@@ -451,7 +451,7 @@ public class ArtistDetailActivity extends AbsSlidingMusicPanelActivity implement
 
     @Override
     public Loader<Artist> onCreateLoader(int id, Bundle args) {
-        return new AsyncArtistDataLoader(this, args.getInt(EXTRA_ARTIST_ID));
+        return new AsyncArtistDataLoader(this, args.getLong(EXTRA_ARTIST_ID));
     }
 
     @Override
@@ -467,9 +467,9 @@ public class ArtistDetailActivity extends AbsSlidingMusicPanelActivity implement
     }
 
     private static class AsyncArtistDataLoader extends WrappedAsyncTaskLoader<Artist> {
-        private final int artistId;
+        private final long artistId;
 
-        public AsyncArtistDataLoader(Context context, int artistId) {
+        public AsyncArtistDataLoader(Context context, long artistId) {
             super(context);
             this.artistId = artistId;
         }

--- a/app/src/main/java/com/kabouzeid/gramophone/ui/activities/MainActivity.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/ui/activities/MainActivity.java
@@ -293,7 +293,7 @@ public class MainActivity extends AbsSlidingMusicPanelActivity {
             MusicPlayerRemote.playFromUri(uri);
             handled = true;
         } else if (MediaStore.Audio.Playlists.CONTENT_TYPE.equals(mimeType)) {
-            final int id = (int) parseIdFromIntent(intent, "playlistId", "playlist");
+            final long id = parseIdFromIntent(intent, "playlistId", "playlist");
             if (id >= 0) {
                 int position = intent.getIntExtra("position", 0);
                 List<Song> songs = new ArrayList<>(PlaylistSongLoader.getPlaylistSongList(this, id));
@@ -301,14 +301,14 @@ public class MainActivity extends AbsSlidingMusicPanelActivity {
                 handled = true;
             }
         } else if (MediaStore.Audio.Albums.CONTENT_TYPE.equals(mimeType)) {
-            final int id = (int) parseIdFromIntent(intent, "albumId", "album");
+            final long id = parseIdFromIntent(intent, "albumId", "album");
             if (id >= 0) {
                 int position = intent.getIntExtra("position", 0);
                 MusicPlayerRemote.openQueue(AlbumLoader.getAlbum(this, id).songs, position, true);
                 handled = true;
             }
         } else if (MediaStore.Audio.Artists.CONTENT_TYPE.equals(mimeType)) {
-            final int id = (int) parseIdFromIntent(intent, "artistId", "artist");
+            final long id = parseIdFromIntent(intent, "artistId", "artist");
             if (id >= 0) {
                 int position = intent.getIntExtra("position", 0);
                 MusicPlayerRemote.openQueue(ArtistLoader.getArtist(this, id).getSongs(), position, true);

--- a/app/src/main/java/com/kabouzeid/gramophone/util/MusicUtil.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/util/MusicUtil.java
@@ -44,13 +44,13 @@ import java.util.regex.Pattern;
  */
 public class MusicUtil {
 
-    public static Uri getMediaStoreAlbumCoverUri(int albumId) {
+    public static Uri getMediaStoreAlbumCoverUri(long albumId) {
         final Uri sArtworkUri = Uri.parse("content://media/external/audio/albumart");
 
         return ContentUris.withAppendedId(sArtworkUri, albumId);
     }
 
-    public static Uri getSongFileUri(int songId) {
+    public static Uri getSongFileUri(long songId) {
         return ContentUris.withAppendedId(MediaStore.Audio.Media.EXTERNAL_CONTENT_URI, songId);
     }
 

--- a/app/src/main/java/com/kabouzeid/gramophone/util/NavigationUtil.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/util/NavigationUtil.java
@@ -24,7 +24,7 @@ import com.kabouzeid.gramophone.ui.activities.PlaylistDetailActivity;
  */
 public class NavigationUtil {
 
-    public static void goToArtist(@NonNull final Activity activity, final int artistId, @Nullable Pair... sharedElements) {
+    public static void goToArtist(@NonNull final Activity activity, final long artistId, @Nullable Pair... sharedElements) {
         final Intent intent = new Intent(activity, ArtistDetailActivity.class);
         intent.putExtra(ArtistDetailActivity.EXTRA_ARTIST_ID, artistId);
 
@@ -36,7 +36,7 @@ public class NavigationUtil {
         }
     }
 
-    public static void goToAlbum(@NonNull final Activity activity, final int albumId, @Nullable Pair... sharedElements) {
+    public static void goToAlbum(@NonNull final Activity activity, final long albumId, @Nullable Pair... sharedElements) {
         final Intent intent = new Intent(activity, AlbumDetailActivity.class);
         intent.putExtra(AlbumDetailActivity.EXTRA_ALBUM_ID, albumId);
 

--- a/app/src/main/java/com/kabouzeid/gramophone/util/PlaylistsUtil.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/util/PlaylistsUtil.java
@@ -184,7 +184,7 @@ public class PlaylistsUtil {
         }
     }
 
-    public static boolean doPlaylistContains(@NonNull final Context context, final long playlistId, final int songId) {
+    public static boolean doPlaylistContains(@NonNull final Context context, final long playlistId, final long songId) {
         if (playlistId != -1) {
             try {
                 Cursor c = context.getContentResolver().query(

--- a/app/src/main/java/com/kabouzeid/gramophone/util/PlaylistsUtil.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/util/PlaylistsUtil.java
@@ -30,7 +30,7 @@ import static android.provider.MediaStore.Audio.Playlists.EXTERNAL_CONTENT_URI;
  */
 public class PlaylistsUtil {
 
-    public static boolean doesPlaylistExist(@NonNull final Context context, final int playlistId) {
+    public static boolean doesPlaylistExist(@NonNull final Context context, final long playlistId) {
         return playlistId != -1 && doesPlaylistExist(context,
                 MediaStore.Audio.Playlists._ID + "=?",
                 new String[]{String.valueOf(playlistId)});

--- a/app/src/main/java/com/kabouzeid/gramophone/util/PlaylistsUtil.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/util/PlaylistsUtil.java
@@ -17,6 +17,7 @@ import com.kabouzeid.gramophone.helper.M3UWriter;
 import com.kabouzeid.gramophone.model.Playlist;
 import com.kabouzeid.gramophone.model.PlaylistSong;
 import com.kabouzeid.gramophone.model.Song;
+import com.kabouzeid.gramophone.service.MusicService;
 
 import java.io.File;
 import java.io.IOException;
@@ -70,6 +71,7 @@ public class PlaylistsUtil {
                 if (cursor != null) {
                     cursor.close();
                 }
+                context.getContentResolver().notifyChange(EXTERNAL_CONTENT_URI, null);
             } catch (SecurityException ignored) {
             }
         }
@@ -92,6 +94,7 @@ public class PlaylistsUtil {
         selection.append(")");
         try {
             context.getContentResolver().delete(EXTERNAL_CONTENT_URI, selection.toString(), null);
+            context.getContentResolver().notifyChange(EXTERNAL_CONTENT_URI, null);
         } catch (SecurityException ignored) {
         }
     }
@@ -133,6 +136,7 @@ public class PlaylistsUtil {
                 Toast.makeText(context, context.getResources().getString(
                         R.string.inserted_x_songs_into_playlist_x, numInserted, getNameForPlaylist(context, playlistId)), Toast.LENGTH_SHORT).show();
             }
+            context.getContentResolver().notifyChange(uri, null);
         } catch (SecurityException ignored) {
         }
     }
@@ -161,6 +165,7 @@ public class PlaylistsUtil {
 
         try {
             context.getContentResolver().delete(uri, selection, selectionArgs);
+            context.getContentResolver().notifyChange(uri, null);
         } catch (SecurityException ignored) {
         }
     }
@@ -180,6 +185,7 @@ public class PlaylistsUtil {
 
         try {
             context.getContentResolver().delete(uri, selection, selectionArgs);
+            context.getContentResolver().notifyChange(uri, null);
         } catch (SecurityException ignored) {
         }
     }
@@ -203,8 +209,10 @@ public class PlaylistsUtil {
     }
 
     public static boolean moveItem(@NonNull final Context context, long playlistId, int from, int to) {
-        return MediaStore.Audio.Playlists.Members.moveItem(context.getContentResolver(),
+        boolean ret = MediaStore.Audio.Playlists.Members.moveItem(context.getContentResolver(),
                 playlistId, from, to);
+        context.getContentResolver().notifyChange(EXTERNAL_CONTENT_URI, null);
+        return ret;
     }
 
     public static void renamePlaylist(@NonNull final Context context, final long id, final String newName) {
@@ -215,6 +223,7 @@ public class PlaylistsUtil {
                     contentValues,
                     MediaStore.Audio.Playlists._ID + "=?",
                     new String[]{String.valueOf(id)});
+            context.getContentResolver().notifyChange(EXTERNAL_CONTENT_URI, null);
         } catch (SecurityException ignored) {
         }
     }

--- a/app/src/main/java/com/kabouzeid/gramophone/util/PlaylistsUtil.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/util/PlaylistsUtil.java
@@ -42,8 +42,8 @@ public class PlaylistsUtil {
                 new String[]{name});
     }
 
-    public static int createPlaylist(@NonNull final Context context, @Nullable final String name) {
-        int id = -1;
+    public static long createPlaylist(@NonNull final Context context, @Nullable final String name) {
+        long id = -1;
         if (name != null && name.length() > 0) {
             try {
                 Cursor cursor = context.getContentResolver().query(EXTERNAL_CONTENT_URI,
@@ -59,12 +59,12 @@ public class PlaylistsUtil {
                     if (uri != null) {
                         Toast.makeText(context, context.getResources().getString(
                                 R.string.created_playlist_x, name), Toast.LENGTH_SHORT).show();
-                        id = Integer.parseInt(uri.getLastPathSegment());
+                        id = Long.parseLong(uri.getLastPathSegment());
                     }
                 } else {
                     // Playlist exists
                     if (cursor.moveToFirst()) {
-                        id = cursor.getInt(cursor.getColumnIndex(MediaStore.Audio.Playlists._ID));
+                        id = cursor.getLong(cursor.getColumnIndex(MediaStore.Audio.Playlists._ID));
                     }
                 }
                 if (cursor != null) {
@@ -96,13 +96,13 @@ public class PlaylistsUtil {
         }
     }
 
-    public static void addToPlaylist(@NonNull final Context context, final Song song, final int playlistId, final boolean showToastOnFinish) {
+    public static void addToPlaylist(@NonNull final Context context, final Song song, final long playlistId, final boolean showToastOnFinish) {
         List<Song> helperList = new ArrayList<>();
         helperList.add(song);
         addToPlaylist(context, helperList, playlistId, showToastOnFinish);
     }
 
-    public static void addToPlaylist(@NonNull final Context context, @NonNull final List<Song> songs, final int playlistId, final boolean showToastOnFinish) {
+    public static void addToPlaylist(@NonNull final Context context, @NonNull final List<Song> songs, final long playlistId, final boolean showToastOnFinish) {
         final int size = songs.size();
         final ContentResolver resolver = context.getContentResolver();
         final String[] projection = new String[]{
@@ -153,7 +153,7 @@ public class PlaylistsUtil {
         return contentValues;
     }
 
-    public static void removeFromPlaylist(@NonNull final Context context, @NonNull final Song song, int playlistId) {
+    public static void removeFromPlaylist(@NonNull final Context context, @NonNull final Song song, long playlistId) {
         Uri uri = MediaStore.Audio.Playlists.Members.getContentUri(
                 "external", playlistId);
         String selection = MediaStore.Audio.Playlists.Members.AUDIO_ID + " =?";
@@ -166,7 +166,7 @@ public class PlaylistsUtil {
     }
 
     public static void removeFromPlaylist(@NonNull final Context context, @NonNull final List<PlaylistSong> songs) {
-        final int playlistId = songs.get(0).playlistId;
+        final long playlistId = songs.get(0).playlistId;
         Uri uri = MediaStore.Audio.Playlists.Members.getContentUri(
                 "external", playlistId);
         String selectionArgs[] = new String[songs.size()];
@@ -202,7 +202,7 @@ public class PlaylistsUtil {
         return false;
     }
 
-    public static boolean moveItem(@NonNull final Context context, int playlistId, int from, int to) {
+    public static boolean moveItem(@NonNull final Context context, long playlistId, int from, int to) {
         return MediaStore.Audio.Playlists.Members.moveItem(context.getContentResolver(),
                 playlistId, from, to);
     }

--- a/app/src/main/java/com/kabouzeid/gramophone/util/RingtoneManager.java
+++ b/app/src/main/java/com/kabouzeid/gramophone/util/RingtoneManager.java
@@ -41,7 +41,7 @@ public class RingtoneManager {
                 .show();
     }
 
-    public   void setRingtone(@NonNull final Context context, final int id) {
+    public   void setRingtone(@NonNull final Context context, final long id) {
         final ContentResolver resolver = context.getContentResolver();
         final Uri uri = MusicUtil.getSongFileUri(id);
         try {

